hw/xbox/xbox.c

```diff
@@ -250,31 +250,38 @@ uint8_t *load_eeprom(void)
             return NULL;
         }
 
-        fd = open(filename, O_RDONLY | O_BINARY);
-        if (fd < 0) {
+        eeprom_fd = open(filename, O_RDWR | O_BINARY);
+        if (eeprom_fd < 0) {
             fprintf(stderr, "qemu: EEPROM file '%s' could not be opened.\n", filename);
             g_free(filename);
             exit(1);
             return NULL;
         }
 
-        rc = read(fd, eeprom_data, eeprom_size);
+        rc = read(eeprom_fd, eeprom_data, eeprom_size);
         if (rc != eeprom_size) {
             fprintf(stderr, "qemu: Could not read the full EEPROM file.\n");
-            close(fd);
+            close(eeprom_fd);
             g_free(filename);
             exit(1);
             return NULL;
         }
 
-        close(fd);
         g_free(filename);
     } else {
         memcpy(eeprom_data, default_eeprom, eeprom_size);
     }
     return eeprom_data;
 }
 
+void save_eeprom(uint8_t *eepromBuffer)
```

Comment:
Parameter name still bad. Not sure about a better one, but if you want to keep it, this should still be `eeprom_buffer`